README.md

```diff
@@ -63,1498 +104,468 @@ Delegating a capability is like the statement `message`. Task is akin to `messag
-
-[Task]s listed in the `fork` field are first-class and only ordered by promises; they otherwise SHOULD be considered independent and equal. As such, atomic guarantees such as failure of one effect implying failure of other effects if left undefined.
-
-The `join` field describes an OPTIONAL "special" [Invocation] which instruct the [Executor] that the [Task] [Invocation] is a continuation of the previous Invocation. This roughly emulates a virtual thread which terminates in an Invocation that produces Effect without a `join` field.
-
-Tasks in the `fork` field MAY be related to the Task in the `join` field if there exists a Promise referencing either Task. If such a promise does not exist, then they SHOULD be treated as entirely separate and MAY be scheduled, deferred, fail, retry, and so on entirely separately.
-
-## 7.1 Schema
-
-```ipldsch
-# Represents a request to invoke enclosed set of tasks concurrently
-type Effects {
-  # Primary set of tasks to be invoked
-  fork      [&Invocation]
-  
-  # Additional task to be invoked with added semantics
-  # of representing a workflow execution continuation.
-  join      optional &Invocation
-}
+})
 ```
 
-## 7.2 Fields
-
-### 7.2.1 Forked Task Invocations
+### Agents
+[Agents]: #agents
 
-The OPTIONAL `fork` field, if present MUST be a list of an alphabetically ordered [Task] links. List MUST NOT not contain duplicate entries.
+#### Issuer
+[Issuer]: #issuer
 
-### 7.2.2 Joined Task Invocation
+The `iss` field MUST include the Issuer of the Invocation. This DID MUST match against the encoding signature.
 
-The OPTIONAL `join` field, if present MUST be set to a [Task] link.
+#### Subject
+[Subject]: #subject
 
-## 7.3 DAG-JSON Examples
+The REQUIRED `sub` field both parameterizes over a specific agent, and acts as a namespace for how to interpret the [Command]. This is especially critical for two parts of the life cycle:
 
-### 7.3.1 Effect spawning concurrent threads
-
-```json
-{
-  "fork": [
-    {
-      "/": "bafyreigmmdzix2vxboojvv6j6h7sgvxnrecdxtglwtqpxw7hybebzlsax4"
-    },
-    {
-      "/": "bafyreif6gfpzgxnii4ys6a4bjenefg737fb5bgam3onrbmhnoa4llk244q"
-    }
-  ]
-}
-```
+1. Specifying a particular `sub` (and thus `aud`) when [enqueuing new Tasks][enqueue] in a Receipt
+2. Indexing Receipts for reverse lookup and memoization
 
-### 7.3.2 Effect continuing thread execution
+#### Audience
+[Audience]: #audience
 
-```json
-{
-  "join": {
-    "/": "bafyreievhy7rnzot7mnzbnqtiajhxx7fyn7y2wkjtuzwtmnflty3767dny"
-  }
-}
-```
+The OPTIONAL `aud` field specified the intended recipient of Invocation, otherwise the Audience MUST be assumed to the [Subject]. This is useful for message routing, command brokers, proxy execution, gateways, replicated state machines, and so on.
```

Comment:
But `aud` is still [REQUIRED in the high-level UCAN spec](https://github.com/ucan-wg/spec/blob/9b50374ba600eea56a697f204f870cd36fba37c7/README.md#payload)?